src/main/java/edu/cornell/kfs/coa/businessobject/SubAccountGlobal.java

```diff
@@ -38,28 +62,130 @@ public SubAccountGlobal() {
+			}
+
+			// update finReportOrganizationCode
+			if (StringUtils.isNotBlank(finReportOrganizationCode)) {
+				subAccount.setFinReportOrganizationCode(finReportOrganizationCode);
+			}
+
+			// update financialReportingCode
+			if (StringUtils.isNotBlank(financialReportingCode)) {
+				subAccount.setFinancialReportingCode(financialReportingCode);
+			}
+
+		    // update indirectCostRecoveryTypeCode
+		    if(StringUtils.isNotBlank(a21SubAccount.indirectCostRecoveryTypeCode)){
+		    	subAccount.getA21SubAccount().setIndirectCostRecoveryTypeCode(a21SubAccount.indirectCostRecoveryTypeCode);
+		    }
+		    
+		    // update financialIcrSeriesIdentifier
+		    if(StringUtils.isNotBlank(a21SubAccount.financialIcrSeriesIdentifier)){
+		    	subAccount.getA21SubAccount().setFinancialIcrSeriesIdentifier(a21SubAccount.financialIcrSeriesIdentifier);
+		    }
+		    
+		    // update costShareChartOfAccountCode
+		    if(StringUtils.isNotBlank(a21SubAccount.costShareChartOfAccountCode)){
+		    	subAccount.getA21SubAccount().setCostShareChartOfAccountCode(a21SubAccount.costShareChartOfAccountCode);
+		    }
+		    
+		    // update costShareSourceAccountNumber
+		    if(StringUtils.isNotBlank(a21SubAccount.costShareSourceAccountNumber)){
+		    	subAccount.getA21SubAccount().setCostShareSourceAccountNumber(a21SubAccount.costShareSourceAccountNumber);
+		    }
+		    
+		    // update costShareSourceSubAccountNumber
+		    if(StringUtils.isNotBlank(a21SubAccount.costShareSourceSubAccountNumber)){
+		    	subAccount.getA21SubAccount().setCostShareSourceSubAccountNumber(a21SubAccount.costShareSourceSubAccountNumber);
+		    }
+		    
+    		// update icr account collections
+		    List<A21IndirectCostRecoveryAccount> newA21IcrAccounts = new ArrayList<A21IndirectCostRecoveryAccount>();
+		    List<A21IndirectCostRecoveryAccount> a21IcrAccounts = subAccountGlobalDetail.getSubAccount().getA21SubAccount().getA21IndirectCostRecoveryAccounts();
+		    
+		    boolean found = false;
+			if (ObjectUtils.isNotNull(indirectCostRecoveryAccounts) && indirectCostRecoveryAccounts.size() > 0) {
+				for (IndirectCostRecoveryAccountChange icrAccount : indirectCostRecoveryAccounts) {
+					subAccountGlobalDetail.refreshReferenceObject("subAccount");
+					found = false;
+					if (ObjectUtils.isNotNull(a21IcrAccounts) && a21IcrAccounts.size() > 0) {
+						for (A21IndirectCostRecoveryAccount a21ICRAccount : a21IcrAccounts) {
+							if (icrAccount.getIndirectCostRecoveryFinCoaCode().equalsIgnoreCase(a21ICRAccount.getIndirectCostRecoveryFinCoaCode()) && icrAccount.getIndirectCostRecoveryAccountNumber().equalsIgnoreCase(a21ICRAccount.getIndirectCostRecoveryAccountNumber())) {
+								if ((icrAccount.getAccountLinePercent().compareTo(a21ICRAccount.getAccountLinePercent()) == 0) && (icrAccount.isActive() != a21ICRAccount.isActive())) {
+									a21ICRAccount.setActive(icrAccount.isActive());
+									found = true;
+									break;
```

Comment:
Not sure if using a "break" statement in a "for" loop is ideal, although a more iterative version of this would require use of an index variable or an Iterator.